annotations/views/repository_views.py

```diff
@@ -239,18 +241,83 @@ def repository_collection_texts(request, repository_id, group_id, group_collecti
         return render(request, 'annotations/repository_ioerror.html', {'error': str(e)}, status=500)
 
     project_id = request.GET.get('project_id', None)
+    
+    # Pagination
+    paginator = Paginator(texts['items'], settings.PAGINATION_PAGE_SIZE)
+    
+    page = request.GET.get('page', 1)
+    try:
+        paginated_texts = paginator.page(page)
+    except PageNotAnInteger:
+        paginated_texts = paginator.page(1)
+    except EmptyPage:
+        paginated_texts = paginator.page(paginator.num_pages)
+
     context = {
         'user': user,
         'repository': repository,
-        'texts': texts['items'],
+        'texts': paginated_texts,
         'title': f'Texts in Collection: ', # Collection name is rendered from frontend
-        'group_info':texts['group'],
-        'group_id':group_id,
+        'group_info': texts['group'],
+        'group_id': group_id,
         'project_id': project_id,
     }
 
     return render(request, 'annotations/repository_collections_text_list.html', context)
 
+@citesphere_authenticated
+def repository_group_texts(request, repository_id, group_id):
+    """View to fetch and display texts within a group from Citesphere"""
+    user = request.user
+    repository = get_object_or_404(Repository, pk=repository_id)
+    manager = RepositoryManager(user=user, repository=repository)
+
+    project_id = request.GET.get('project_id', None)
+    page = request.GET.get('page', 1)
+    try:
+        page = int(page)
+    except ValueError:
+        page = 1
+
+    items_per_page = getattr(settings, 'PAGINATION_PAGE_SIZE')
+
+    try:
+        texts = manager.group_items(group_id, page=page, items_per_page=items_per_page)
+    except Exception as e:
+        return render(request, 'annotations/repository_ioerror.html', {'error': str(e)}, status=500)
+
+    # Pagination
+    total_items = texts['total_items'] 
+    items = texts['items']
+
+    # Create a Paginator object with a dummy list to represent total items
+    # range(total_items) is being used as a placeholder to create correct pagination metadata
+    paginator = Paginator(range(total_items), items_per_page)
+    try:
+        paginated_texts = paginator.page(page)
+    except PageNotAnInteger:
+        page = 1
+        paginated_texts = paginator.page(page)
+    except EmptyPage:
+        page = paginator.num_pages
+        paginated_texts = paginator.page(page)
```

Comment:
I'm confused, why is this needed? If the user asks for a page that doesn't exist or is not a number, citesphere should either throw an error or return an empty list, so why would you need to check this here again? (in other words, I don't think you need a paginator object?)